src/braket/experimental_capabilities/experimental_capability_context.py

```diff
@@ -24,40 +34,110 @@ class ExperimentalCapabilityContextError(Exception):
 class GlobalExperimentalCapabilityContext:
     def __init__(self) -> None:
         """A global singleton that tracks whether experimental capabilities are enabled."""
-        self._is_enabled = False
+        self._all_enabled = False
+        self._currently_enabled_capabilities = set()
 
-    @property
-    def is_enabled(self) -> bool:
-        """Check if experimental capabilities are enabled.
+    def is_enabled(
+        self,
+        experimental_capabilities: ExperimentalCapability
+        | Iterable[ExperimentalCapability]
+        | None = None,
+    ) -> bool:
+        """Check if experimental capabilities are enabled, either for all or a specific
+        set of experimental capabilities.
 
         Returns:
             bool: True if experimental capabilities are enabled, False otherwise.
         """
-        return self._is_enabled
+        return self._is_enabled(experimental_capabilities)
 
-    def enable(self) -> None:
-        """Enable all experimental capabilities."""
-        self._is_enabled = True
+    @singledispatchmethod
+    def _is_enabled(
+        self,
+        experimental_capabilities: Any,
+    ):
+        raise TypeError(
+            "If provided, Experimental capabilities must be a single \
+                or list of ExperimentalCapability strings"
+        )
+
+    @_is_enabled.register
+    def _(self, experimental_capabilities: None):
```

Comment:
This is ambiguous that None implies ALL here. Should we make  `_is_enabled(ALL)` explicit?